### PR TITLE
Modify platform check in USBGuard rules

### LINKED ISSUE #8248: Remediations for usbguard do not all have conditionals to regard containers as not applicable.
#### Description of problem:
Remediations for usbguard do not all have conditionals to ignore containers. When generating the remediation script, it includes installing usbguard and this will cause automated builds to fail. 

USBguard is actually not included in the RHEL UBI8 Repositories and it will not install. 

I apologize if this is the wrong space for this, if someone could please redirect me if possible.

#### SCAP Security Guide Version:
scap-security-guide-0.1.61

#### Operating System Version:
RHEL UBI8 8.5 (All RHEL UBI8's)

#### Steps to Reproduce:
1. wget -O ssg-rhel8-ds-1.2.xml https://github.com/RedHatGov/rhel8-stig-latest/blob/master/ssg-rhel8-ds-1.2.xml
2. oscap xccdf eval --profile xccdf_org.ssgproject.content_profile_stig --results-arf arf.xml --fetch-remote-resources ssg-rhel8-ds-1.2.xml
3. oscap xccdf generate fix --result-id xccdf_org.open-scap_testresult_xccdf_org.ssgproject.content_profile_stig --fix-type bash arf.xml > remediation.sh
3. Run the remediation

#### Actual Results:
yum install -y "usbguard"
Red Hat Universal Base Image 8 (RPMs) - BaseOS                                                                                                                          649 kB/s | 796 kB     00:01
Red Hat Universal Base Image 8 (RPMs) - AppStream                                                                                                                       1.9 MB/s | 2.6 MB     00:01
Red Hat Universal Base Image 8 (RPMs) - CodeReady Builder                                                                                                                19 kB/s |  16 kB     00:00
No match for argument: usbguard
Error: Unable to find a match: usbguard

This causes builds on RHEL UBI8 to fail

#### Expected Results:
Applicable remediations are applied successfully (usbguard is not applicable to containers and is skipped) and build continues.

#### Additional Information/Debugging Steps:
In the fix section of ssg-rhel8-ds-1.2.xml where it's building the remediation script. 2 of the 7 usbguard related scripts have containers as not applicable.

Incomplete Logic:
BASH = if ! grep -q s390x /proc/sys/kernel/osrelease; then

Expected Logic:
BASH = if ! grep -q s390x /proc/sys/kernel/osrelease &amp;&amp; { [ ! -f /.dockerenv ] &amp;&amp; [ ! -f /run/.containerenv ]; }; then
Ansible = ansible_virtualization_type not in ["docker", "lxc", "openvz", "podman", "container"]

Missing from:
xccdf_org.ssgproject.content_rule_package_usbguard_installed
xccdf_org.ssgproject.content_rule_configure_usbguard_auditbackend
xccdf_org.ssgproject.content_rule_usbguard_allow_hid
xccdf_org.ssgproject.content_rule_usbguard_allow_hid_and_hub
xccdf_org.ssgproject.content_rule_usbguard_allow_hub

Included in:
xccdf_org.ssgproject.content_rule_service_usbguard_enabled 
xccdf_org.ssgproject.content_rule_usbguard_generate_policy

### PR DESCRIPTION
**Pull request by cortesana**
#### Description:

This PR introduces additional conditions for USBGuard rules' checks and remediations to ignore containers.
This change will mean that USBGuard rules will result in `notapplicable` when on containers, therefore the installation and configuration of USBGuard will not be included in the remediation script and will prevent automatic builds from failing.

#### Rationale:

USBGuard rules should not be applicable to containers in any distribution.

Fixes #8248

#### Review Hints:
Some comments have been included in order to facilitate the PR review.

### CODE REVIEW

File: linux_os/guide/services/usbguard/group.yml
Comment (cortesana) [inline linux_os/guide/services/usbguard/group.yml:8]: As USBGuard rules should not be applicable to containers, I have included the additional conditions here, rather than the individual `rule.yml` files.

**Comment by jan-cerny** [comment]
@cortesana congratulations!

please don't assign your own PRs to yourself

**Comment by jan-cerny** [comment]
@cortesana  We're sorry but we had an issue with the CI jobs and unfortunately they're required to pass before merging. Could you please rebase this PR on the top of the latest upstream master branch? That will include fix from https://github.com/ComplianceAsCode/content/pull/10250 to this PR and will unblock the CI.

**Review by marcusburghardt — APPROVED** [review]
All rules in the `USBGuard` group are passing in VMs.
The `platform` entry was removed from all rules, except `configure_usbguard_auditbackend`, and properly centralized in the `group.yml`

Thanks @cortesana

**Pull request merged**
